common/src/main/java/com/microsoft/identity/common/internal/broker/InactiveBrokerClient.java

```diff
@@ -0,0 +1,166 @@
+package com.microsoft.identity.common.internal.broker;
+
+import android.content.Context;
+import android.content.Intent;
+import android.os.Bundle;
+import android.os.RemoteException;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import com.microsoft.aad.adal.IBrokerAccountService;
+import com.microsoft.identity.common.adal.internal.AuthenticationConstants;
+import com.microsoft.identity.common.exception.BaseException;
+import com.microsoft.identity.common.exception.ClientException;
+import com.microsoft.identity.common.internal.logging.Logger;
+
+import java.util.concurrent.ExecutionException;
+
+/**
+ * Use for communicating with inactive broker only.
+ * This is NOT being used in ADAL. If you plan to do so, make sure it works properly.
+ **/
+public class InactiveBrokerClient {
+    private static final String TAG = InactiveBrokerClient.class.getSimpleName();
+
+    private static final String BROKER_ACCOUNT_SERVICE_INTENT_FILTER = "com.microsoft.workaccount.BrokerAccount";
+    private static final String BROKER_ACCOUNT_SERVICE_CLASS_NAME = "com.microsoft.aad.adal.BrokerAccountService";
+
+    private Context mContext;
+    private String mCallingPackageName;
+    private BrokerAccountServiceConnection mBrokerAccountServiceConnection;
+    private Intent mBrokerAccountServiceIntent;
+    private Boolean mBound = false;
+
+    // Must be called from BG thread.
+    public static String getSerializedKeyFromInactiveBroker(@NonNull Context context,
+                                                            @NonNull final String callingPackageName) {
+        final String methodName = ":getKeyFromInactiveBroker";
+
+        final InactiveBrokerClient client = new InactiveBrokerClient(context, callingPackageName);
+        try {
+            final BrokerAccountServiceFuture brokerAccountServiceFuture = client.connect();
+            IBrokerAccountService service = brokerAccountServiceFuture.get();
+
+            Bundle resultBundle = service.getInactiveBrokerKey();
+
+            // This means that the inactive broker doesn't support getInactiveBrokerKey().
+            if (resultBundle == null){
+                Logger.verbose(TAG + methodName, "resultBundle is null");
```

Comment:
should be log the package name (which broker i mean)  and also version here  ?

Reply:
If you're referring to the inactive broker's name, it's already logged in getInactiveBrokerPackageName().